deployment/data-streams/utils/txutil/transaction.go

```diff
@@ -0,0 +1,95 @@
+package txutil
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+)
+
+// PreparedTx represents a transaction that was prepared but not sent to the chain. This is intended to be
+// either signed and executed directly or bundled into an MCMS operation.
+type PreparedTx struct {
+	Tx            *gethtypes.Transaction
+	ChainSelector uint64
+	ContractType  string
+	Tags          []string
+}
+
+type ExecuteTxResult struct {
+	Tx          *PreparedTx
+	BlockNumber uint64
+}
+
+// SignAndExecute signs and then executes transactions directly on the chain with the given deployer key configured
+// for the chain. The transactions should not be already sent to the chain.
+func SignAndExecute(e deployment.Environment, preparedTxs []*PreparedTx) ([]ExecuteTxResult, error) {
+	for _, tx := range preparedTxs {
+		chain := e.Chains[tx.ChainSelector]
```

Comment:
Might be good to check that `chain` exists, i.e. it's not `nil`. That might happen if it's not configured for this `env`.

Reply:
good catch I've added the check